app/javascript/packs/components/RestaurantItem.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+function Rating(props) {
```

Comment:
instead of all this you could use `map` to loop over the `props.stars` and do it inline (or still in a function).
btw, you can use the library `loadash` for its convenient `map` function.